platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcCredentialsResolver.java

```diff
@@ -47,50 +50,62 @@
 import org.pac4j.oidc.client.OidcClient;
 import org.pac4j.oidc.config.OidcConfiguration;
 import org.pac4j.oidc.credentials.OidcCredentials;
-import org.pac4j.oidc.credentials.authenticator.OidcAuthenticator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CustomOidcAuthenticator extends OidcAuthenticator {
+public class OidcCredentialsResolver {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(CustomOidcAuthenticator.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(OidcCredentialsResolver.class);
 
+  private OidcConfiguration oidcConfiguration;
+  private OidcClient oidcClient;
   private OidcTokenValidator oidcTokenValidator;
   private OIDCProviderMetadata metadata;
 
-  public CustomOidcAuthenticator(
-      OidcConfiguration configuration, OidcClient client, OIDCProviderMetadata metadata) {
-    super(configuration, client);
+  public OidcCredentialsResolver(
+      OidcConfiguration oidcConfiguration, OidcClient oidcClient, OIDCProviderMetadata metadata) {
+    this.oidcConfiguration = oidcConfiguration;
+    this.oidcClient = oidcClient;
     this.metadata = metadata;
-    oidcTokenValidator = new OidcTokenValidator(configuration, metadata);
+    oidcTokenValidator = new OidcTokenValidator(oidcConfiguration, metadata);
   }
 
   /* This methods job is to try and get an id token from a
   1. refresh token
   2. authorization code
   3. access token
   */
-  @Override
-  public void validate(OidcCredentials credentials, WebContext webContext) {
-    final RefreshToken refreshToken = credentials.getRefreshToken();
-    final AuthorizationCode authorizationCode = credentials.getCode();
+  public void resolveIdToken(OidcCredentials credentials, WebContext webContext) {
+    final AccessToken initialAccessToken = credentials.getAccessToken();
+    final JWT initialIdToken = credentials.getIdToken();
+    if (initialIdToken != null) {
+      if (initialAccessToken != null) {
+        oidcTokenValidator.validateAccessToken(initialAccessToken, initialIdToken);
+      }
+      oidcTokenValidator.validateIdTokens(initialIdToken, webContext);
+      return;
+    }
+
+    final RefreshToken initialRefreshToken = credentials.getRefreshToken();
+    final AuthorizationCode initialAuthorizationCode = credentials.getCode();
 
     final List<AuthorizationGrant> grantList = new ArrayList<>();
 
-    if (refreshToken != null) {
-      grantList.add(new RefreshTokenGrant(refreshToken));
+    if (initialRefreshToken != null) {
+      // TODO add refresh token validation
```

Comment:
Idk about this TODO the thing we don't do anything with it. This validation won't make sense unless we have the ability to exchange a refresh token to an id token as we do with the access token

Reply:
👀 Looks like I never responded to this comment.

I left this todo here as a placeholder for "where do I put refresh token validation logic" future work.